Makefile

```diff
@@ -43,7 +43,7 @@ kubeseal: $(GO_FILES)
 	$(GO) build -o $@ $(GO_FLAGS) -ldflags "$(GO_LD_FLAGS)" ./cmd/kubeseal
 
 %-static: $(GO_FILES)
-	CGO_ENABLED=0 $(GO) build -o $@ -installsuffix cgo $(GO_FLAGS) -ldflags "$(GO_LD_FLAGS)" ./cmd/$*
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(GO) build -o $@ -installsuffix cgo $(GO_FLAGS) -ldflags "$(GO_LD_FLAGS)" ./cmd/$*
```

Comment:
Not sure I understand the idea here. Make doesn't know whether a pre-requisite compiled file was built with the correct flags, unless I am missing something.

Reply:
I explained it poorly (and incorrectly).  I meant this:

```patch
--- a/Makefile
+++ b/Makefile
@@ -45,6 +45,8 @@ kubeseal: $(GO_FILES)
 %-static: $(GO_FILES)
        CGO_ENABLED=0 $(GO) build -o $@ -installsuffix cgo $(GO_FLAGS) -ldflags "$(GO_LD_FLAGS)" ./cmd/$*
 
+docker/controller: export GOOS = linux
+docker/controller: export GOARCH = amd64
 docker/controller: controller-static
        cp $< $@
 
```

It's not bulletproof, but assuming `controller-static` is first built through the `docker/controller` dependency (rather than some other dependency path) then it should get the right GOOS/GOARCH.